src/workerd/util/test.h

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) 2024 Cloudflare, Inc.
+// Licensed under the Apache 2.0 license found in the LICENSE file or at:
+//     https://opensource.org/licenses/Apache-2.0
+
+#pragma once
+#include <kj/test.h>
+
+namespace workerd {
+
+// Checks that code throws an exception matching the type and message of the given exception.
+#define WD_EXPECT_THROW(expException, code, ...)                                                   \
+  do {                                                                                             \
+    auto expExcObj = expException;                                                                 \
+    KJ_IF_SOME(e, ::kj::runCatchingExceptions([&]() { (void)({ code; }); })) {                     \
```

Comment:
Nit: what about exceptions like `jsg::JsExceptionThrown`? If those aren't important to capture here, all good... but might be good to specify that in a comment

Reply:
Good point...  The function does not currently provide special handling for `jsg::JsExceptionThrown`, and I hadn't been fully aware that those exceptions might need to be handled differently.

However, I think the function as-is is sufficient for a test to ensure that a thrown `kj::Exception` has the expected type and message, which is what it is currently used for.  If a `JsExceptionThrown` is thrown instead, it would be captured via `getCaughtExceptionAsKj()` as a `kj::Exception` of type "failed", with an extra `std::exception` in the description.  So I think it would hard at least for a `JsExceptionThrown` to accidentally match an expected non-`JsExceptionThrown` exception.

I've added a comment advising that the macro is intended for matching `kj::Exception`s, and that someone who wants to assert on a JS exception should probably use a separate macro.